code/datums/statistics/random_facts/random_fact.dm

```diff
@@ -10,7 +11,73 @@
 	check_xeno = set_check_xeno
 
 /datum/random_fact/proc/announce()
+	calculate_announcement_message()
 	if(message)
 		to_world(SPAN_CENTERBOLD(message))
 		return TRUE
 	return FALSE
+
+/datum/random_fact/proc/calculate_announcement_message()
+	var/death_stat_gotten = 0
+	var/living_stat_gotten = 0
+	var/datum/entity/statistic/death/death_to_report = null
+	var/mob/mob_to_report = null
+
+	if(round_statistics && length(round_statistics.death_stats_list))
+		for(var/datum/entity/statistic/death/death in round_statistics.death_stats_list)
+			if(!check_human && !death.is_xeno)
+				continue
+			if(!check_xeno && death.is_xeno)
+				continue
+			if(death_stat_gotten < death_grab_stat(death))
+				death_to_report = death
+				death_stat_gotten = death_grab_stat(death)
+
+	var/list/list_to_check = list()
+	if(check_human)
+		list_to_check += GLOB.alive_human_list
+	if(check_xeno)
+		list_to_check += GLOB.living_xeno_list
+
+	for(var/mob/checked_mob as anything in list_to_check)
+		if(living_stat_gotten < life_grab_stat(checked_mob))
+			mob_to_report = checked_mob
+			living_stat_gotten = life_grab_stat(checked_mob)
+
+	if(!death_to_report && !mob_to_report)
+		return
+
+	var/name = ""
+	var/stat_gotten = 0
+	var/additional_message = ""
+	if(death_to_report && mob_to_report)
+		if(living_stat_gotten > death_stat_gotten)
+			name = mob_to_report.real_name
+			stat_gotten = living_stat_gotten
+			additional_message = "and survived! Great work!"
+		else
+			name = death_to_report.mob_name
+			stat_gotten = death_stat_gotten
+			additional_message = "before dying"
+			if(death_to_report.cause_name)
+				additional_message += " to <b>[death_to_report.cause_name]</b>"
+			additional_message += ". Good work!"
+	else if(death_to_report)
+		name = death_to_report.mob_name
+		stat_gotten = death_stat_gotten
+		additional_message = "before dying"
+		if(death_to_report.cause_name)
+			additional_message += " to <b>[death_to_report.cause_name]</b>"
+		additional_message += ". Good work!"
+	else
+		name = mob_to_report.real_name
+		stat_gotten = living_stat_gotten
+		additional_message = "and survived! Great work!"
+
+	message = "<b>[name]</b> earned <b>[stat_gotten] [statistic_name]</b> [additional_message]"
```

Comment:
Earning damage is a bit of a weird way for this to be phrased now: 
![image](https://github.com/cmss13-devs/cmss13/assets/76988376/f57f9c18-ffcd-4c7e-a774-49aeba2e08a6)